### PR TITLE
Children row in inspector props

### LINKED ISSUE #5686: Data Children: show props.children in the inspector component section
**Problem:**
it feels weird to see `name` as a data cartouche in the navigator for `<h2>{name}</h2>`, but the inspector's component section is empty. we want the component section to be able to show anything which might require data picking!

**proposed solution:**
Show props.children as a separate entry at the bottom of the component section!

Question: should we show the component section with props.children for selected elements which would otherwise not show the component section, such as div, h1, h2, etc? I personally feel like yes we should

### PR DESCRIPTION
**Pull request by ruggi**
**Problem:**

The children prop for components is not shown in the inspector.

**Fix:**

- Show the `children` prop in the inspector and allow changing it, via either a cartouche or number/string inputs
- Allow setting children elements when asking for a prop update
- For number and string children, use string/number inputs - or the default JSX component (or cartouche for data references) otherwise
- If the selected component does not support children, don't show it

https://github.com/concrete-utopia/utopia/assets/1081051/a66df108-2812-4c7d-9d65-aa83f8641bc6

**Manual Tests:**
I hereby swear that:

- [x] I opened a hydrogen project and it loaded
- [x] I could navigate to various routes in Preview mode

Fixes #5686

### CODE REVIEW

File: editor/src/components/editor/element-children.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import {
+  isIntrinsicHTMLElement,
+  getJSXElementNameAsString,
+} from '../../core/shared/element-template'
+import type { PropertyControlsInfo } from '../custom-code/code-file'
+import type { EditorAction } from './action-types'
+import { deleteView, replaceElementInScope } from './actions/action-creators'
+import * as EP from '../../core/shared/element-path'
+
+export function elementSupportsChildrenFromPropertyControls(
```
Comment (ruggi) [inline editor/src/components/editor/element-children.ts:18]: verbatim from `navigator-item-components.tsx`

File: editor/src/components/inspector/sections/component-section/property-control-controls.tsx [hunk 2]
```diff
@@ -348,8 +331,32 @@ export const NumberInputPropertyControl = React.memo(
...
+        // if the target prop is children, dig for its value
+        if (propName === 'children') {
+          const element = MetadataUtils.findElementByElementPath(
+            store.editor.jsxMetadata,
+            props.elementPath,
+          )
+          if (element != null && isRight(element.element) && isJSXElement(element.element.value)) {
+            const child = element.element.value.children[0]
+            if (child.type === 'ATTRIBUTE_OTHER_JAVASCRIPT') {
+              const parsed = parseNumber(child.originalJavascript)
```
Comment (ruggi) [inline editor/src/components/inspector/sections/component-section/property-control-controls.tsx:346]: this kinda sucks a bit, if you have better ideas please do tell!

**Review by bkrmendy — APPROVED** [review]

File: editor/src/components/inspector/sections/component-section/property-control-controls.tsx (hunk 2 above)
Comment (balazsbajorics) [reply editor/src/components/inspector/sections/component-section/property-control-controls.tsx:346]: I _think_ the info we are looking for is in the `editorState.allElementProps`! IIRC we store props.children in the spy, it's not filtered out

Comment (balazsbajorics) [reply editor/src/components/inspector/sections/component-section/property-control-controls.tsx:346]: maybe we could even put this into `propMetadata.value` so here you don't have to special case it

**Review by balazsbajorics — APPROVED** [review]

**Review by balazsbajorics — APPROVED** [review]

**Pull request merged**